service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -3,23 +3,49 @@
  */
 package com.lightbend.lagom.javadsl.server;
 
-import com.lightbend.lagom.internal.javadsl.*;
+import com.google.inject.Binder;
+import com.lightbend.lagom.internal.javadsl.BinderAccessor;
+import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServices;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServicesProvider;
 import com.lightbend.lagom.internal.javadsl.server.ServiceInfoProvider;
-import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl;
-
-import com.lightbend.lagom.javadsl.server.status.MetricsService;
-import com.google.inject.Binder;
 import com.lightbend.lagom.javadsl.api.ServiceInfo;
 import com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport;
+import com.lightbend.lagom.javadsl.server.status.MetricsService;
+
+import java.util.Arrays;
+import java.util.Optional;
 
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+    /**
+     * @deprecated use {@link ServiceGuiceSupport#bindServices(String, ServiceBinding[])} instead.
+     */
     default void bindServices(ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.empty(), serviceBindings);
+    }
+
+    /**
+     * binds the provided services and builds the {@link ServiceInfo} for this service.
+     *
+     * @param serviceName
+     * @param serviceBindings
+     */
+    default void bindServices(String serviceName, ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.of(serviceName), serviceBindings);
+    }
```

Comment:
Can't keep both methods since reference will become ambiguous. We can only use a new approach or replace 'bindServices(ServiceBinding<?>...)' with `bindServices(ServiceBinding<?>,ServiceBinding<?>...)`

```
com/lightbend/lagom/discovery/impl/ServiceRegistryModule.java:33: reference to bindServices is ambiguous
[error]   both method bindServices(com.lightbend.lagom.javadsl.server.ServiceGuiceSupport.ServiceBinding<?>...) in com.lightbend.lagom.javadsl.server.ServiceGuiceSupport and method bindServices(com.lightbend.lagom.javadsl.server.ServiceGuiceSupport.ServiceBinding<?>,com.lightbend.lagom.javadsl.server.ServiceGuiceSupport.ServiceBinding<?>...) in com.lightbend.lagom.javadsl.server.ServiceGuiceSupport match
[error]                 bindServices(serviceBinding(ServiceRegistry.class, ServiceRegistryImpl.class));
```

This will break compatibility because `bindServices()` won't be synthesized. Runtime will be fine since invoking `bindServices(...)` without arguments caused an ArrayIndexOutOfBoundsException.